core/model/modx/modresource.class.php

```diff
@@ -450,6 +450,43 @@ function __construct(xPDO & $xpdo) {
         $this->_cacheKey= "[contextKey]/resources/[id]";
     }
 
+    /**
+     * Compute the "preview URL" for the resource
+     *
+     * @return string
+     */
+    public function getPreviewUrl() {
+        $url = '';
+        if (!$this->get('deleted')) {
+            $this->xpdo->setOption('cache_alias_map', false);
+            $sessionEnabled = '';
+            /** @var modContextSetting|null $ctxSetting */
+            $ctxSetting = $this->xpdo->getObject(
+                'modContextSetting',
+                array(
+                    'context_key' => $this->get('context_key'),
+                    'key' => 'session_enabled'
+                )
+            );
+
+            if ($ctxSetting) {
```

Comment:
The missing suggestion:
```suggestion
            if ($ctxSetting && $ctxSetting->get('value') == 0) {
```